catalog/spatial/registry/registry-admin-modules/registry-admin-local-ui/src/main/webapp/js/model/Association.js

```diff
@@ -0,0 +1,250 @@
+        if (segment.get('segmentId') === id) {
+            return segment;
+        }
+        for (var index = 0; index < segment.get('segments').models.length; index++) {
+            var seg = getSegmentModel(segment.get('segments').models[index], id);
+            if (seg) {
+                return seg;
+            }
+        }
+    }
+
+    var Association = {};
+
+    Association.Association = Backbone.AssociatedModel.extend({
+        defaults: {
+            id: undefined,
+            type: undefined,
+            sourceId: undefined,
+            sourceName: undefined,
+            targetId: undefined,
+            targetName: undefined,
+            targetType: undefined
+        },
+        populateFromModel: function (association, topLevelSegment) {
+            if (association) {
+                this.set('id', association.id);
+                this.set('type', association.associationType);
+                this.set('sourceId', association.sourceObject);
+                this.set('targetId', association.targetObject);
+            }
+            var sourceSeg = getSegmentModel(topLevelSegment, this.get('sourceId'));
+            var targetSeg = getSegmentModel(topLevelSegment, this.get('targetId'));
+            this.set('targetType', targetSeg.get('segmentType'));
+            this.set('sourceName', sourceSeg.constructTitle ? sourceSeg.constructTitle() : sourceSeg.getField('Name').get('value'));
+            this.set('targetName', targetSeg.constructTitle ? targetSeg.constructTitle() : targetSeg.getField('Name').get('value'));
+        }
+
+    });
+    Association.Associations = Backbone.Collection.extend({
+        model: Association.Association
+    });
+
+    Association.AssociationModel = Backbone.AssociatedModel.extend({
+        relations: [
+            {
+                type: Backbone.Many,
+                key: 'associations',
+                collectionType: function () {
+                    return Association.Associations;
+                },
+                relatedModel: function () {
+                    return Association.Association;
+                }
+            },
+            {
+                type: Backbone.Many,
+                key: 'associationSegments',
+                collectionType: function () {
+                    return Association.SegmentIds;
+                },
+                relatedModel: function () {
+                    return Association.SegmentId;
+                }
+            },
+            {
+                type: Backbone.One,
+                key: 'topSegment',
+                relatedModel: function () {
+                    return Segment.Segment;
+                }
+            }
+        ],
+        defaults: {
+            associations: [],
+            associationSegments: [],
+            topSegment: null
+        },
+        populateFromModel: function (associations) {
+            var model = this;
+            var collectionModel = this.get('associations');
```

Comment:
Newer versions of javascript do have constants now but it requires new browsers like IE11 so I would not suggest using it here without using a tool to transpile to es5 which is supported by more browsers.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const